ft_virtualchannel.m

```diff
@@ -0,0 +1,382 @@
+    end
+    
+    % check that the spatial filter exists
+    if isempty(source.filter{indx(i)})
+      ft_error(sprintf('the spatial filter information is missing for dipole with position [%d %d %d]', cfg.pos(i,:)));
+    end
+  end
+  
+elseif useparcellation
+  
+  % do some checks and balances on the input data and cfg
+  
+  % keep the transformation matrix
+  if isfield(parcellation, 'transform')
+    transform = parcellation.transform;
+  else
+    transform = [];
+  end
+  
+  % ensure it is a parcellation, not a segmentation
+  parcellation = ft_checkdata(parcellation, 'datatype', 'parcellation', 'parcel lationstyle', 'indexed');
+  
+  % keep the transformation matrix
+  if ~isempty(transform)
+    parcellation.transform = transform;
+  end
+ 
+  % ensure that the source and the parcellation are anatomically consistent
+  if ~isalmostequal(source.pos, parcellation.pos, 'abstol', 1000000*eps)
+    ft_error('the source positions are not consistent with the parcellation, please use FT_SOURCEINTERPOLATE');
+  end
+  
+  if isequal(cfg.parcel, 'all')
+    cfg.parcel = parcellation.(sprintf('%slabel',cfg.parcellation));
+  end
+  
+  indx = match_str(parcellation.(sprintf('%slabel',cfg.parcellation)), cfg.parcel);
+  cfg.parcel = parcellation.(sprintf('%slabel',cfg.parcellation))(indx);
+  
+  nvc = numel(cfg.parcel);
+end
+
+% create a montage for each of the dipoles/parcels, and use ft_apply_montage,
+% followed by a dimensionality reduction step, using ft_componentanalysis
+unmixing  = cell(1, nvc);
+topolabel = cell(1, nvc);
+tmpdata   = cell(1, nvc);
+for i = 1:nvc
+  
+  montage = [];
+  if usepos
+    
+    montage.tra      = source.filter{indx(i)};
+    montage.labelold = source.label;
+    montage.labelnew = cell(size(montage.tra,1), 1);
+    for k = 1:size(montage.tra,1)
+      montage.labelnew{k} = sprintf('virtualchannel%03d_orientation%03d', i, k);
+    end
+    bname = sprintf('virtualchannel%03d', i);
+    
+  elseif useparcellation
+  
+    sel = find(parcellation.(cfg.parcellation)==indx(i));
+    montage.tra      = cat(1, source.filter{sel});
+    montage.labelold = source.label;
+    montage.labelnew = cell(size(montage.tra,1), 1);
+    cnt = 0;
+    for k = 1:numel(sel)
+      for kk = 1:size(source.filter{sel(k)},1)
+        cnt = cnt+1;
+        montage.labelnew{cnt} = sprintf('%s_dipole%03d_orientation%03d', cfg.parcel{i}, k, kk);
+      end
+    end
+    bname = matlab.lang.makeValidName(cfg.parcel{i}); % FIXME not sure how universal this is.
+    
+  end
+  
+  % apply the montage to the numeric data, but make an exception for the
+  % svd method: this one is much more efficient, in that its implementattion
+  % operates on the covariance matrix directly, bypasses ft_componentanalysis, 
+  % and does not require the intermediate (possibly memory greedy, and slow)
+  % data representation.
+  if ~isequal(cfg.method, 'svd')
```

Comment:
Might be clearer with `if isequal(cfg.method, 'svd')` first as the special case, and with the default case last in the `else` block.

Reply:
good suggestion, thanks